platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -203,27 +213,259 @@ As a result, `toolChains` should now be configured directly at the top-level of
+This is to help prevent misconfigurations where the tests are written for one test framework but the test task is mistakenly configured to use another test framework.
+If filters are applied, the outcome depends on the `failOnNoMatchingTests` property.
 
-The default version of TestNG is link:https://github.com/testng-team/testng/releases/tag/7.11.0[7.11.0].
+This behavior can be disabled by setting the `failOnNoDiscoveredTests` property to `false` in the test task configuration:
 
-The default version of Spock is link:https://spockframework.org/spock/docs/2.3/release_notes.html[2.3].
+====
+[.multi-language-sample]
+=====
+.build.gradle.kts
+[source,kotlin]
+----
+tasks.withType<AbstractTestTask>().configureEach {
+    failOnNoDiscoveredTests = false
+}
+----
+=====
+[.multi-language-sample]
+=====
+.build.gradle
+[source,groovy]
+----
+tasks.withType(AbstractTestTask).configureEach {
+    failOnNoDiscoveredTests = false
+}
+----
+=====
+====
+
+==== Stale outputs outside the build directory are no longer deleted
+
+In previous versions of Gradle, class files located outside the build directory were deleted when considered stale. This was a special case for class files registered as outputs of a source set.
+
+Because this setup is uncommon and forced Gradle to eagerly realize all compile related tasks in every build, the behavior has been removed in Gradle 9.
+
+Gradle will continue to clean up stale outputs inside the build directory as needed.
+
+==== `model` and `component` tasks are no longer automatically added
+
+The `model` and `component` tasks report on the structure of legacy software model objects configured for the project.
+Previously, these tasks were automatically added to the project for every build.
+These tasks are now only added to a project when a rule-based plugin is applied (such as those provided by Gradle's support for building native software).
+
+=== Configuration Cache changes
+
+==== Unsupported build event listeners are now configuration cache problems
+
+The build event listener registration method
+link:{javadocPath}/org/gradle/build/event/BuildEventsListenerRegistry.html#onTaskCompletion(org.gradle.api.provider.Provider)[`BuildEventsListenerRegistry.onTaskCompletion`]
+accepts arbitrary providers of any
+link:{javadocPath}/org/gradle/tooling/events/OperationCompletionListener.html[`OperationCompletionListener`]
+implementations.
+However, only providers returned from
+link:{javadocPath}/org/gradle/api/services/BuildServiceRegistry.html#registerIfAbsent(java.lang.String,java.lang.Class,org.gradle.api.Action)[`BuildServiceRegistry.registerIfAbsent`]
+or
+link:{javadocPath}/org/gradle/api/services/BuildServiceRegistration.html#getService()[`BuildServiceRegistration.getService`]
+are currently supported when the Configuration Cache is enabled.
+
+Previously, unsupported providers were silently discarded and never received events when the configuration cache was used.
```

Comment:
Is it still a case when graceful degradation is enabled? Why should we fail the build instead of degrading in such a case? CC @6hundreds